File: 03-frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from './address';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

export class Purchase {
  custormer: Customer;
  shippingAddress: Address;
  billingAddress: Address;
  order: Order;
  orderItems: OrderItem[];
}

File: 02-backend/spring-boot-ecommerce/src/main/java/com/learn/ecommerce/dto/Purchase.java
package com.learn.ecommerce.dto;

import java.util.Set;

import com.learn.ecommerce.entity.Address;
import com.learn.ecommerce.entity.Customer;
import com.learn.ecommerce.entity.Order;
import com.learn.ecommerce.entity.OrderItem;

import lombok.Data;

@Data
public class Purchase {

	private Customer customer;
	private Address shippingAddress;
	private Address billingAddress;
	private Order order;
	private Set<OrderItem> orderItems;
}
